it/src/test/java/org/corfudb/universe/scenario/NodeUpAndPartitionedIT.java

```diff
@@ -0,0 +1,106 @@
+     * 3) Create two link failures between a responsive node with smaller endpoint name and the
+     * rest of the cluster AND restart the unresponsive node.
+
+     * 4) Verify that responsive node mentioned in step 3 becomes unresponsive
+     * 5) Verify that the restarted unresponsive node in step 3 gets healed
+     * 6) Verify cluster status and data path
+     */
+    @Test(timeout = 300000)
+    public void nodeUpAndPartitionedTest() {
+        getScenario().describe((fixture, testCase) -> {
+            CorfuCluster corfuCluster = universe.getGroup(fixture.getCorfuCluster().getName());
+
+            CorfuClient corfuClient = corfuCluster.getLocalCorfuClient();
+
+            CorfuTable<String, String> table = corfuClient.createDefaultCorfuTable(DEFAULT_STREAM_NAME);
+            for (int i = 0; i < DEFAULT_TABLE_ITER; i++) {
+                table.put(String.valueOf(i), String.valueOf(i));
+            }
+
+            testCase.it("Should fail the node with most link failures to unresponsive set",
+                    data -> {
+                        // Step 1)
+                        CorfuServer server0 = corfuCluster.getServerByIndex(0);
+                        CorfuServer server1 = corfuCluster.getServerByIndex(1);
+                        CorfuServer server2 = corfuCluster.getServerByIndex(2);
+
+                        // Step 2)
+                        server1.stop(Duration.ofSeconds(10));
+                        waitForUnresponsiveServersChange(size -> size == 1, corfuClient);
+
+                        assertThat(corfuClient.getLayout().getUnresponsiveServers())
+                                .containsExactly(server1.getEndpoint());
+
+                        // Step 3) Partition the responsive server0 from both unresponsive server1
+                        // and responsive server2 and reconnect server 1. Wait for layout's unresponsive
+                        // servers to change After this, cluster becomes unavailable.
+                        // NOTE: cannot use waitForClusterDown() since the partition only happens on server side, client
+                        // can still connect to two nodes, write to table so system down handler will not be triggered.
+                        server0.disconnect(Collections.singletonList(server1));
+                        server0.disconnect(Collections.singletonList(server2));
+                        server1.start();
+                        waitForLayoutChange(layout -> layout.getUnresponsiveServers()
+                                                            .contains(server0.getEndpoint()),
+                                            corfuClient);
+
+                        // Step 4)
+                        assertThat(corfuClient.getLayout().getUnresponsiveServers())
+                                .contains(server0.getEndpoint());
+
+
+                        // Step 5)
+                        waitForUnresponsiveServersChange(size -> size == 1, corfuClient);
```

Comment:
There will be a time that there are potentially two unresponsive nodes before one of them getting healed. This step is to make sure that the one which could be healed is actually healed.

Reply:
Oh I didn't notice you are using `contain` instead of `containsExactly`. But why don't we combine all these checks:

```
waitForLayoutChange(layout -> layout.getUnresponsiveServers()
                                       .contains(server0.getEndpoint()),
                                       corfuClient);

 // Verify server0 is unresponsive
assertThat(corfuClient.getLayout().getUnresponsiveServers())
                        .contains(server0.getEndpoint());

// Verify unresponsive server1 gets healed
waitForUnresponsiveServersChange(size -> size == 1, corfuClient);
assertThat(corfuClient.getLayout().getUnresponsiveServers())
                        .containsExactly(server0.getEndpoint());
``` 

into two: 
```
waitForUnresponsiveServersChange(size -> size == 1, corfuClient);
assertThat(corfuClient.getLayout().getUnresponsiveServers()).containsExactly(server0.getEndpoint());
```

which means  we wait until it stabilizes and do a final check?

Reply:
Let's keep it as is so that in addition to checking the stabile result, we also assert that first step is adding the correct node to unresponsive set.